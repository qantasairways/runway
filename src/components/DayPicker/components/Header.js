import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';

import {
  colours,
  fontFamily,
  fontSize,
  mq,
  layout
} from '../../../theme/airways';

import Tabs from './Tabs';
import QantasLogo from '../../../icons/QantasLogo';
import CrossIcon from '../../../icons/CrossIcon';

const ICON_SIZE = '32px';

const weekdayStyles = {
  width: '100%',
  height: '30px',
  fontFamily: fontFamily.body,
  fontSize: '0.875rem',
  lineHeight: '0.875rem',
  textAlign: 'center',
  boxShadow: '0 1px 1px 0 rgba(0, 0, 0, 0.08), 0 0 1px 0 rgba(0, 0, 0, 0.04)',
  [mq.medium]: {
    height: '40px'
  }
};

function orderLabels(firstDayOfWeek, dayLabels) {
  return dayLabels
    .slice(firstDayOfWeek)
    .concat(dayLabels.slice(0, firstDayOfWeek));
}

function Header({
  closeDialog,
  startDate,
  endDate,
  firstDayOfWeek,
  isSelectingStartDate,
  isDateRange,
  closeAriaLabel,
  headerLabel,
  dayLabels,
  startSelectedLabel,
  endSelectedLabel,
  startPlaceholder,
  endPlaceholder,
  rowStyles,
  Icon
}) {
  const orderedLabels = orderLabels(firstDayOfWeek, dayLabels);

  return (
    <div css={{ zIndex: 1 }}>
      <div
        css={{
          fontFamily: fontFamily.bold,
          background: colours.darkerGrey,
          color: colours.white
        }}
      >
        <div
          css={{
            display: 'flex',
            justifyContent: 'space-between',
            alignItems: 'center',
            maxWidth: layout.containerMaxWidth,
            margin: '0 auto',
            padding: '0 0 0 12px'
          }}
        >
          <QantasLogo
            className={css({
              display: 'none',
              width: '126px',
              [mq.medium]: {
                display: 'block'
              }
            })}
          />
          <span
            css={{
              fontSize: fontSize.body,
              lineHeight: 0,
              [mq.medium]: {
                paddingRight: '71px',
                fontSize: fontSize.labelLarge
              }
            }}
          >
            {headerLabel}
          </span>
          <button
            aria-label={closeAriaLabel}
            onClick={closeDialog}
            type="button"
            css={{
              height: '48px',
              width: '48px',
              padding: 0,
              lineHeight: 0,
              background: 'none',
              border: 'none',
              cursor: 'pointer',
              [mq.medium]: {
                height: '55px',
                width: '55px'
              },
              [mq.large]: {
                textAlign: 'right'
              }
            }}
          >
            <CrossIcon
              css={{ fill: 'white' }}
              height={ICON_SIZE}
              width={ICON_SIZE}
            />
          </button>
        </div>
      </div>
      <Tabs
        isSelectingStartDate={isSelectingStartDate}
        startPlaceholder={startPlaceholder}
        endPlaceholder={endPlaceholder}
        startSelectedLabel={startSelectedLabel}
        endSelectedLabel={endSelectedLabel}
        startDate={startDate && startDate.toDateString()}
        endDate={endDate && endDate.toDateString()}
        isDateRange={isDateRange}
        Icon={Icon}
      />
      <div
        css={{
          background: colours.disabledGrey
        }}
      >
        <div css={{ ...rowStyles, ...weekdayStyles }}>
          {orderedLabels.map(day => (
            <div
              key={day}
              css={{
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                position: 'relative',
                '&::after': {
                  content: '""',
                  position: 'absolute',
                  right: '-1px',
                  height: '15px',
                  borderLeft: '1px solid #eeeeee'
                },
                [mq.xLarge]: {
                  fontSize: fontSize.body
                }
              }}
            >
              {day}
            </div>
          ))}
        </div>
      </div>
    </div>
  );
}

Header.propTypes = {
  closeDialog: PropTypes.func.isRequired,
  startDate: PropTypes.instanceOf(Date),
  endDate: PropTypes.instanceOf(Date),
  isSelectingStartDate: PropTypes.bool,
  isDateRange: PropTypes.bool,
  firstDayOfWeek: PropTypes.number.isRequired,
  closeAriaLabel: PropTypes.string,
  headerLabel: PropTypes.string,
  startSelectedLabel: PropTypes.string,
  endSelectedLabel: PropTypes.string,
  startPlaceholder: PropTypes.string,
  endPlaceholder: PropTypes.string,
  dayLabels: PropTypes.arrayOf(PropTypes.string).isRequired,
  rowStyles: PropTypes.shape().isRequired,
  Icon: PropTypes.func
};

Header.defaultProps = {
  startDate: null,
  endDate: null,
  isSelectingStartDate: true,
  isDateRange: false,
  startSelectedLabel: '',
  endSelectedLabel: '',
  startPlaceholder: '',
  endPlaceholder: '',
  closeAriaLabel: '',
  headerLabel: '',
  Icon: null
};

export default Header;
