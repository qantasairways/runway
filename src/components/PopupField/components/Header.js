import React from 'react';
import PropTypes from 'prop-types';

import { colours, fontFamily } from '../../../theme/airways';

import CrossIcon from '../../../icons/CrossIcon';

const ICON_SIZE = '32px';

function Header({ closePopup, closeAriaLabel, headerLabel, HeaderIcon }) {
  return (
    <div css={{ zIndex: 1 }}>
      <div
        css={{
          fontFamily: fontFamily.bold,
          background: colours.darkerGrey,
          color: colours.white,
          display: 'flex',
          justifyContent: 'space-between',
          alignItems: 'center',
          padding: '0 0 0 12px'
        }}
      >
        {HeaderIcon && (
          <HeaderIcon
            height={ICON_SIZE}
            width={ICON_SIZE}
            css={{ fill: colours.white, marginRight: '8px' }}
          />
        )}
        <span
          css={{
            flex: 1,
            justifySelf: 'left'
          }}
        >
          {headerLabel}
        </span>
        <button
          aria-label={closeAriaLabel}
          onClick={closePopup}
          type="button"
          css={{
            background: 'none',
            height: `65px`,
            padding: 0,
            lineHeight: 0,
            border: 'none',
            width: '52px',
            cursor: 'pointer'
          }}
        >
          <CrossIcon
            css={{ fill: 'white' }}
            height={ICON_SIZE}
            width={ICON_SIZE}
          />
        </button>
      </div>
    </div>
  );
}

Header.propTypes = {
  closePopup: PropTypes.func.isRequired,
  closeAriaLabel: PropTypes.string,
  headerLabel: PropTypes.string,
  HeaderIcon: PropTypes.func
};

Header.defaultProps = {
  closeAriaLabel: '',
  headerLabel: '',
  HeaderIcon: null
};

export default Header;
