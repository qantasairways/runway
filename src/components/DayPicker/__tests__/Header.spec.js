import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';
import noop from '../../../utils/noop';

const start = new Date(2019, 3, 21, 0, 0, 0, 0);
const end = new Date(2019, 3, 24, 0, 0, 0, 0);
const CalendarIcon = () => <div>Calendar Icon</div>;

describe('Header', () => {
  let component;

  it('renders correctly with defaults', () => {
    component = shallow(
      <Header
        closeDialog={() => {}}
        firstDayOfWeek={1}
        rowStyles={{}}
        dayLabels={['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']}
        onKeyDown={noop}
        setFocusElementRef={noop}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with props provided', () => {
    component = shallow(
      <Header
        headerLabel="header label"
        firstDayOfWeek={5}
        closeAriaLabel="close"
        closeDialog={() => {}}
        rowStyles={{}}
        dayLabels={['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']}
        startDate={start}
        endDate={end}
        isSelectingStartDate={false}
        isSingleDate
        startSelectedLabel="depart"
        endSelectedLabel="return"
        startPlaceholder="depart when?"
        endPlaceholder="return when?"
        Icon={CalendarIcon}
        isDateRange={false}
        onKeyDown={noop}
        setFocusElementRef={noop}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
