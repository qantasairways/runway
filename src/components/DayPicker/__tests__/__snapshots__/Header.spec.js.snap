// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly with defaults 1`] = `
.emotion-12 {
  z-index: 1;
}

.emotion-3 {
  font-family: "Ciutadella-Md",sans-serif;
  background: #323232;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0 0 12px;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  justify-self: left;
}

.emotion-2 {
  background: none;
  height: 48px;
  padding: 0;
  line-height: 0;
  border: none;
  width: 52px;
  cursor: pointer;
}

.emotion-1 {
  fill: white;
}

.emotion-11 {
  width: 100%;
  padding: 8px 0;
  font-family: "Ciutadella",sans-serif;
  font-size: 0.875rem;
  line-height: 0.875rem;
  text-align: center;
  background: #fafafa;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.08),0 0 1px 0 rgba(0,0,0,0.04);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-4::after {
  content: "";
  position: absolute;
  right: -1px;
  height: 15px;
  border-left: 1px solid #eeeeee;
}

<div
  className="emotion-12"
>
  <div
    className="emotion-3"
  >
    <span
      className="emotion-0"
    />
    <button
      aria-label=""
      className="emotion-2"
      onClick={[Function]}
      type="button"
    >
      <CrossIcon
        className="emotion-1"
        height="32px"
        width="32px"
      />
    </button>
  </div>
  <div
    className="emotion-11"
  >
    <div
      className="emotion-4"
      key="Tue"
    >
      Tue
    </div>
    <div
      className="emotion-4"
      key="Wed"
    >
      Wed
    </div>
    <div
      className="emotion-4"
      key="Thu"
    >
      Thu
    </div>
    <div
      className="emotion-4"
      key="Fri"
    >
      Fri
    </div>
    <div
      className="emotion-4"
      key="Sat"
    >
      Sat
    </div>
    <div
      className="emotion-4"
      key="Sun"
    >
      Sun
    </div>
    <div
      className="emotion-4"
      key="Mon"
    >
      Mon
    </div>
  </div>
</div>
`;

exports[`Header renders correctly with props provided 1`] = `
.emotion-12 {
  z-index: 1;
}

.emotion-3 {
  font-family: "Ciutadella-Md",sans-serif;
  background: #323232;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0 0 12px;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  justify-self: left;
}

.emotion-2 {
  background: none;
  height: 48px;
  padding: 0;
  line-height: 0;
  border: none;
  width: 52px;
  cursor: pointer;
}

.emotion-1 {
  fill: white;
}

.emotion-11 {
  width: 100%;
  padding: 8px 0;
  font-family: "Ciutadella",sans-serif;
  font-size: 0.875rem;
  line-height: 0.875rem;
  text-align: center;
  background: #fafafa;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.08),0 0 1px 0 rgba(0,0,0,0.04);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-4::after {
  content: "";
  position: absolute;
  right: -1px;
  height: 15px;
  border-left: 1px solid #eeeeee;
}

<div
  className="emotion-12"
>
  <div
    className="emotion-3"
  >
    <span
      className="emotion-0"
    >
      header label
    </span>
    <button
      aria-label="close"
      className="emotion-2"
      onClick={[Function]}
      type="button"
    >
      <CrossIcon
        className="emotion-1"
        height="32px"
        width="32px"
      />
    </button>
  </div>
  <div
    className="emotion-11"
  >
    <div
      className="emotion-4"
      key="Sat"
    >
      Sat
    </div>
    <div
      className="emotion-4"
      key="Sun"
    >
      Sun
    </div>
    <div
      className="emotion-4"
      key="Mon"
    >
      Mon
    </div>
    <div
      className="emotion-4"
      key="Tue"
    >
      Tue
    </div>
    <div
      className="emotion-4"
      key="Wed"
    >
      Wed
    </div>
    <div
      className="emotion-4"
      key="Thu"
    >
      Thu
    </div>
    <div
      className="emotion-4"
      key="Fri"
    >
      Fri
    </div>
  </div>
</div>
`;
